orcid-core/src/main/java/org/orcid/core/web/filters/CrossDomainWebManger.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * =============================================================================
+ *
+ * ORCID (R) Open Source
+ * http://orcid.org
+ *
+ * Copyright (c) 2012-2014 ORCID, Inc.
+ * Licensed under an MIT-Style License (MIT)
+ * http://orcid.org/open-source-license
+ *
+ * This copyright and license information (including a link to the full license)
+ * shall be included in its entirety in all copies or substantial portion of
+ * the software.
+ *
+ * =============================================================================
+ */
+package org.orcid.core.web.filters;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.orcid.pojo.ajaxForm.PojoUtil;
+import org.springframework.beans.factory.annotation.Value;
+
+public class CrossDomainWebManger {
+
+    @Value("${org.orcid.security.cors.allowed_domains:orcid.org}")
+    private String allowedDomains;
+    
+    @Value("${org.orcid.security.cors.allowed_path:/public}")
+    private String allowedPaths;
+    
+    @Value("${org.orcid.security.cors.allowed_functions}")
+    private String allowedFunctions;    
+    
+    private List<String> domainsRegex;  
+    
+    public boolean allowed(HttpServletRequest request) throws MalformedURLException {
+        URL url = new URL(request.getRequestURL().toString());
+        String path = url.getPath();        
+        
+        //Check origin header
+        if(!PojoUtil.isEmpty(request.getHeader("origin"))) {
```

Comment:
I would more likely make it a && check. If origin is null and referrer is localhost.